internal/snmp/field.go

```diff
@@ -166,11 +172,11 @@ func (f *Field) Convert(ent gosnmp.SnmpPDU) (v interface{}, err error) {
 		case uint64:
 			v = int64(vt)
 		case []byte:
-			v, _ = strconv.ParseInt(string(vt), 10, 64)
+			v, err = strconv.ParseInt(string(vt), 10, 64)
 		case string:
-			v, _ = strconv.ParseInt(vt, 10, 64)
+			v, err = strconv.ParseInt(vt, 10, 64)
 		}
-		return v, nil
+		return v, err
```

Comment:
I'm still not certain this is the right thing. This err is declared above on line 104 right? So this could be non-nil when we finally return and I don't think that is the intention.

Am I reading this right?

Reply:
I don't think you are right. The `err` in line 104 has a local scope to the `if` branch and is thus not used here. That's why I hate explicit naming in return variables as it makes things more complex! 

Anyway, despite the code being correct I would, either explicitly declare the error here (and remove the naming in the return values) **or** do the error checking in the  `[]byte` and `string` branch to ease reading the code.